convert2rhel/checks.py

```diff
@@ -768,6 +768,71 @@ def is_loaded_kernel_latest():
             _, build_time, latest_kernel, repoid = tuple(str(line).split("\t"))
             packages.append((build_time, latest_kernel, repoid))
+    # Repoquery doesn't return any text at all when it can't find any matches for the query (when used with --quiet)
+    if len(repoquery_output) > 0:
+        # Convert to an tuple split with `buildtime` and `kernel` version.
+        # We are also detecting if the sentence "listed more than once in the configuration"
+        # appears in the repoquery output. If it does, we ignore it.
+        # This later check is supposed to avoid duplicate repofiles being defined in the system,
+        # this is a super corner case and should not happen everytime, but if it does, we are aware now.
+        packages = [
+            tuple(str(line).split("\t"))
+            for line in repoquery_output.split("\n")
+            if (line.strip() and "listed more than once in the configuration" not in line.lower())
+        ]
+        # Filter out any error messages (things that do not have the three
+        # fields which we expect)
+        packages = [pkg for pkg in packages if len(pkg) == 3]
+
+        # Sort out for the most recent kernel with reverse order
+        # In case `repoquery` returns more than one kernel in the output
+        # We display the latest one to the user.
+        packages.sort(key=lambda x: x[0], reverse=True)
+
+        _, latest_kernel, repoid = packages[0]
+
+        # The loaded kernel version
+        uname_output, _ = run_subprocess(["uname", "-r"], print_output=False)
+        loaded_kernel = uname_output.rsplit(".", 1)[0]
+        match = compare_package_versions(latest_kernel, str(loaded_kernel))
+
+        if match == 0:
+            logger.info("The currently loaded kernel is at the latest version.")
+        else:
+            repos_message = (
+                "in the enabled system repositories"
+                if not reposdir
+                else "in repositories defined in the %s folder" % reposdir
+            )
+            logger.critical(
+                "The version of the loaded kernel is different from the latest version %s.\n"
+                " Latest kernel version available in %s: %s\n"
+                " Loaded kernel version: %s\n\n"
+                "To proceed with the conversion, update the kernel version by executing the following step:\n\n"
+                "1. yum install %s-%s -y\n"
+                "2. reboot" % (repos_message, repoid, latest_kernel, loaded_kernel, package_to_check, latest_kernel)
+            )
+    else:
+        # Repoquery failed to detected any kernel or kernel-core packages in it's repositories
+        # we allow the user to provide a environment variable to override the functionality and proceed
+        # with the conversion, otherwise, we just throw an critical logging to them.
+        allow_older_envvar_names = (
+            "CONVERT2RHEL_ALLOW_SKIP_KERNEL_CURRENCY_CHECK",
```

Comment:
```suggestion
            "CONVERT2RHEL_SKIP_KERNEL_CURRENCY_CHECK",
```
It seems to me that the "ALLOW" is not needed there. WDYT, @Andrew-ang9, @r0x0d, @abadger, @SpyTec?

Reply:
Agreed. Speaking of, I think that the word `currency` here doesn't make too much sense too? 

Maybe it could be: `CONVERT2RHEL_SKIP_KERNEL_CHECK`

Reply:
`CONVERT2RHEL_SKIP_KERNEL_CHECK` would be too broad. Apart from checking whether the kernel is up-to-date, we still check:
- the kernel signature,
- whether the kernel major version corresponds to the RHEL major version,
- whether the kernel name does not include subscrings like `uek`.

The `CURRENCY` sounds ok to me. As an alternative we could use "up-to-date" but that would be quite long (`CONVERT2RHEL_SKIP_KERNEL_UP_TO_DATE_CHECK`).

@mportman12, could you please help us out here?

Reply:
Yeah I would definitely agree that the "Allow" is not needed! I wasn't sure if I should change it so I just left it as is. I will temporally change it to "CONVERT2RHEL_SKIP_KERNEL_CURRENCY_CHECK", until we get a final verdict on this!